### ISSUE TITLE
cool

### ISSUE THREAD

**Issue by inikishev** [flancast90/Harpsichord#1]
any examples

**Comment by flancast90** [comment]
Sure @idkman2021, assuming you were able to follow the Python and pip install instructions, you should be able to run the Harpsichord.py file inside the folder “src.” From there, the program will ask you to input a directory to the code file, and will then convert! 

in case you are talking about examples in the sense of a demo, there are currently no demos available online, since running Python via html would be too time-consuming a process for a simple program. 

let me know if this was helpful, or what I can do to help more, and drop a star if you found the project interesting!

**Comment by inikishev** [comment]
> Sure @idkman2021, assuming you were able to follow the Python and pip install instructions, you should be able to run the Harpsichord.py file inside the folder “src.” From there, the program will ask you to input a directory to the code file, and will then convert!
> 
> in case you are talking about examples in the sense of a demo, there are currently no demos available online, since running Python via html would be too time-consuming a process for a simple program.
> 
> let me know if this was helpful, or what I can do to help more, and drop a star if you found the project interesting!

are there any mp3s that this app generated

**Comment by inikishev** [comment]
i can't follow the python pip instuction installation because i have neither of those three.

**Comment by inikishev** [comment]
> > Sure @idkman2021, assuming you were able to follow the Python and pip install instructions, you should be able to run the Harpsichord.py file inside the folder “src.” From there, the program will ask you to input a directory to the code file, and will then convert!
> > in case you are talking about examples in the sense of a demo, there are currently no demos available online, since running Python via html would be too time-consuming a process for a simple program.
> > let me know if this was helpful, or what I can do to help more, and drop a star if you found the project interesting!
> 
> are there any mp3s that this app generated

that is a demo to see what this app can do so that I can decide whether I should bother trying to follow the instructions with my monkey brain

**Comment by flancast90** [comment]
@idkman2021, for right now, there are no mp3 files, but I'll go ahead and make one. While I'm at it, I'll go ahead and push a new version that makes better-sounding music to impress you more :)

**Comment by flancast90** [comment]
Here you go @idkman2021: https://drive.google.com/file/d/1rXySc2Y7byHt6gHhIgTSLNNTLR1AxJZY/view?usp=sharing. As you can hear, the music is not exactly "harmonious", but I can imagine it being used in a game setting such as the "Mario Castle." 

Hopefully you weren't expecting a full-on Sonata lol, but hopefully we can agree its pretty decent for a computer program and random values! The concept was actually partially inspired by Mozart's Dice Game piece (https://en.wikipedia.org/wiki/Musikalisches_W%C3%BCrfelspiel#:~:text=A%20Musikalisches%20W%C3%BCrfelspiel%20(German%20for,generate%20music%20from%20precomposed%20options.) which used a similar random concept of dice rolls.

**Comment by inikishev** [comment]
I counted 15 notes, is it generated from a code with 15 symbols? I was kinda expecting a faster and more complex piece where structures would emerge from the code it is generated from, 15 very long notes won't lead to any emergence, it just sounds random

**Comment by flancast90** [comment]
Yes @idkman2021, the piece does get a lot faster later on, but for storage reasons, I just recorded the first few notes. 

You're right in that it is a bit random: the program relies on distribution of special characters in the program, combined with a sort of dice roll to pick a plausible note. 

Again, please don't expect the program to generate pitch-perfect music with harmonious chords, etc. the program is a bit random and more of an "is it possible" than anything else. 

If you are looking for the former, you are probably best off looking into music AI's/Neural Networks. If you are interested, I have one in-progress as a private repo right now: when it is done, I could demo it to you to show its capabilities.
